biz.aQute.repository/src/aQute/bnd/repository/maven/provider/IndexFile.java

```diff
@@ -338,17 +339,25 @@ private boolean isMulti(String name) {
 	}
 
 	private Map<Archive, Resource> parseMulti(Archive archive, File multi) throws Exception {
+		Set<String> multis = new HashSet<String>();
+		File localArchives = repo.toLocalFile(archive)
+			.getParentFile();
 		Map<Archive, Resource> result = new HashMap<>();
 		try (Jar jar = new Jar(multi)) {
-			int n = 1000;
 			for (Entry<String, aQute.bnd.osgi.Resource> entry : jar.getResources()
 				.entrySet()) {
 				String path = entry.getKey()
 					.toLowerCase();
 				if (path.endsWith(".jar")) {
+					String name = Strings.lastPathSegment(path)[1];
+					name = Strings.extension(name)[0];
+					name = ZipUtil.cleanPath(name);
+					name = name.replaceAll("[.,;\"'-]+", "_");
```

Comment:
You should precompile the regex into a static final Pattern field. There is no point in compiling the regex each iteration of the loop.